src/modules/cmdpal/Exts/Microsoft.CmdPal.Ext.Registry/Helpers/QueryHelper.cs

```diff
@@ -0,0 +1,113 @@
+﻿// Copyright (c) Microsoft Corporation
+// The Microsoft Corporation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+using Microsoft.CmdPal.Ext.Registry.Constants;
+
+namespace Microsoft.CmdPal.Ext.Registry.Helpers;
+
+/// <summary>
+/// Helper class to easier work with queries
+/// </summary>
+internal static class QueryHelper
+{
+    /// <summary>
+    /// The character to distinguish if the search query contain multiple parts (typically "\\")
+    /// </summary>
+    internal const string QuerySplitCharacter = "\\\\";
+
+    /// <summary>
+    /// A list that contain short names of all registry base keys
+    /// </summary>
+    private static readonly IReadOnlyDictionary<string, string> _shortBaseKeys = new Dictionary<string, string>(6)
+    {
+        { Win32.Registry.ClassesRoot.Name, KeyName.ClassRootShort },
+        { Win32.Registry.CurrentConfig.Name, KeyName.CurrentConfigShort },
+        { Win32.Registry.CurrentUser.Name, KeyName.CurrentUserShort },
+        { Win32.Registry.LocalMachine.Name, KeyName.LocalMachineShort },
+        { Win32.Registry.PerformanceData.Name, KeyName.PerformanceDataShort },
+        { Win32.Registry.Users.Name, KeyName.UsersShort },
+    };
+
+    /// <summary>
+    /// Sanitize the query to avoid issues with the regex
+    /// </summary>
+    /// <param name="query">Query containing front-slash</param>
+    /// <returns>A string replacing all the front-slashes with back-slashes</returns>
+    private static string SanitizeQuery(in string query)
+    {
+        var sanitizedQuery = Regex.Replace(query, @"/(?<=^(?:[^""]*""[^""]*"")*[^""]*)(?<!//.+)", "\\");
```

Comment:
nit: Using GeneratedRegex attribute has better performance

Reply:
This came from the original PT run code, good catch. TIL about `[GeneratedRegex]`